src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc

```diff
@@ -267,10 +267,8 @@ tsi_result alts_tsi_handshaker_result_create(grpc_gcp_HandshakerResp* resp,
   }
   upb_strview local_service_account =
       grpc_gcp_Identity_service_account(local_identity);
-  if (local_service_account.size == 0) {
-    gpr_log(GPR_ERROR, "Invalid local service account");
-    return TSI_FAILED_PRECONDITION;
-  }
+  // we don't check if local service account is empty here
+  // since portable handshake service doesn't have local identity
```

Comment:
I would rephrase as 

"We don't check if local service account is empty here because local identity could be empty in certain situations."